src/db/collections/status.rs

```diff
@@ -24,14 +25,43 @@ impl MongoDb {
             .collection::<StatusDocument>(StatusDocument::COLLECTION)
             .find_one(doc! {}, None)
             .await
-            .map(|doc| doc.map(|doc| doc.network_name))
+            .map(|doc| doc.and_then(|doc| doc.network_name))
     }
 
     /// Sets the name of the network.
     pub async fn set_network_name(&self, network_name: String) -> Result<(), Error> {
         self.0
             .collection::<StatusDocument>(StatusDocument::COLLECTION)
-            .insert_one(StatusDocument { network_name }, None)
+            .update_one(
+                doc! {},
+                doc! { "$set": { "network_name": network_name } },
+                UpdateOptions::builder().upsert(true).build(),
+            )
+            .await?;
```

Comment:
Isn't that a problem that this `StatusDocument` can be changed non-atomically? What if I run `Chronicle` in a different network, but the ledger index is still the old? So what I am asking is ... should setting the network name not reset the ledger index?

Reply:
right, should this method fail then?